source/slang/slang.cpp

```diff
@@ -468,6 +467,15 @@ Linkage::Linkage(Session* session, ASTBuilder* astBuilder)
     m_defaultSourceManager.initialize(session->getBuiltinSourceManager(), nullptr);
 
     setFileSystem(nullptr);
+
+    // Copy of the built in linkages modules
+    if (builtinLinkage)
+    {
+        for (const auto& pair : builtinLinkage->mapNameToLoadedModules)
+        {
+            mapNameToLoadedModules.Add(pair.Key, pair.Value);
+        }
+    }
```

Comment:
Hmm... I'm not sure I like the idea of the module names from the builtin linkage being automatically pulled into any child linkages, but I also can't pinpoint a case where that would cause problems.

I guess one note here is that this change would make it possible for a user file to write `import hlsl;` and have that be interpreted as an import of the module from `hlsl.meta.slang` that is part of the stdlib. At the same time, we don't actually treat the visibility of symbols from `hlsl` or `core` in user code as them having being imported, so doing an import like that would likely create an error for the user. That's probably fine, so long as our stdlib module names are unlikely to conflict with user-defined modules.

I'd have a slight preference here for having any code that wants to search for a module by name first look in the current linkage and then in whatever builtin linkage that refers to, but I'm also fine with this solution as a kind of quick fix.